packages/cloud-functions/src/cico/fetchProviders.ts

```diff
@@ -0,0 +1,124 @@
+import * as functions from 'firebase-functions'
+import { DigitalAsset, FiatCurrency } from '../config'
+import { composeProviderUrl } from './composeProviderUrls'
+import { getProviderAvailability } from './providerAvailability'
+import Simplex, { SimplexQuote } from './Simplex'
+
+export interface UserLocationData {
+  country: string | null
+  state: string | null
+  ipAddress: string | null
+}
+
+export interface ProviderRequestData {
+  userLocation: UserLocationData
+  walletAddress: string
+  fiatCurrency: FiatCurrency
+  digitalAsset: DigitalAsset
+  fiatAmount?: number
+  digitalAssetAmount?: number
+}
+
+export enum Providers {
+  Moonpay = 'Moonpay',
+  Ramp = 'Ramp',
+  Transak = 'Transak',
+  Simplex = 'Simplex',
+  Xanpool = 'Xanpool',
+}
+
+export enum PaymentMethod {
+  Card = 'Card',
+  Bank = 'Bank',
+}
+
+export interface Provider {
+  name: Providers
+  restricted: boolean
+  unavailable?: boolean
+  paymentMethods: PaymentMethod[]
+  url?: string
+  logo: string
+  quote?: SimplexQuote
+  cashIn: boolean
+  cashOut: boolean
+}
+
+export const fetchProviders = functions.https.onRequest(async (request, response) => {
```

Comment:
This looks amazing! Would this allow us to add a new provider without doing any changes to the app?

Reply:
Yes! Everything is happening server-side + using Firebase RTDB so no code changes in Valora are needed :)